src/component/mxgraph/shape/activity-shapes.ts

```diff
@@ -34,20 +34,30 @@ export abstract class BaseActivityShape extends mxRectangleShape {
 
   public paintForeground(c: mxAbstractCanvas2D, x: number, y: number, w: number, h: number): void {
     super.paintForeground(c, x, y, w, h);
-    this.paintMarkerIcon(buildPaintParameter(c, x, y, w, h, this, 0.17, false, 1.5));
-  }
-
-  protected paintMarkerIcon(paintParameter: PaintParameter): void {
-    switch (StyleUtils.getBpmnMarker(this.style)) {
-      case ShapeBpmnMarkerKind.LOOP:
-        this.iconPainter.paintLoopIcon(paintParameter);
-        break;
+    this.paintMarkerIcons(c, x, y, w, h);
+  }
+
+  protected paintMarkerIcons(c: mxAbstractCanvas2D, x: number, y: number, w: number, h: number): void {
+    const markers = StyleUtils.getBpmnMarkers(this.style);
+    if (markers) {
+      markers.split(',').forEach(marker => {
+        switch (marker) {
+          case ShapeBpmnMarkerKind.LOOP:
+            const paintParameter = buildPaintParameter(c, x, y, w, h, this, 0.17, false, 1.5);
+            this.iconPainter.paintLoopIcon(paintParameter);
+            // TODO remove as this is to avoid side effects if the iconPainter has changed colors. So this should be managed in the iconPainter implementation
+            // currently, we don't provide any way to do it (we could pass the original style)
+            // Restore original configuration
+            // TODO missing mxShape.configureCanvas in mxgraph-type-definitions (this will replace explicit function calls)
+            // this.configureCanvas(c, x, y, w, h);
+            paintParameter.c.setStrokeColor(StyleUtils.getStrokeColor(this.style));
```

Comment:
I think we need to put it just after the switch, because we need it for all markers :)

Reply:
ok let's do it